Assets/MixedRealityToolkit.SDK/Features/Utilities/Solvers/TapToPlace.cs

```diff
@@ -0,0 +1,344 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Input;
+using Microsoft.MixedReality.Toolkit.Physics;
+using Microsoft.MixedReality.Toolkit.Utilities.Solvers;
+using System.Runtime.CompilerServices;
+using UnityEngine;
+using UnityEngine.Events;
+
+[assembly: InternalsVisibleTo("Microsoft.MixedReality.Toolkit.SDK.Inspectors")]
+namespace Microsoft.MixedReality.Toolkit.Utilities.Editor.Solvers
+{
+    /// <summary>
+    /// Tap to place is a far interaction component used to place objects on a surface.
+    /// </summary>
+    public class TapToPlace : Solver, IMixedRealityPointerHandler
+    {
+        
+        [SerializeField]
+        [Tooltip("The default distance (in meters) an object will be placed relative to the TrackedTargetType forward in the SolverHandler." +
+            " The GameObjectToPlace will be placed at the default placement distance if a surface is not hit by the raycast.")]
+        private float defaultPlacementDistance = 1.5f;
+
+        /// <summary>
+        /// The default distance (in meters) an object will be placed relative to the TrackedTargetType forward in the SolverHandler.
+        /// The GameObjectToPlace will be placed at the default placement distance if a surface is not hit by the raycast.
+        /// </summary>
+        public float DefaultPlacementDistance
+        {
+            get => defaultPlacementDistance;
+            set => defaultPlacementDistance = value;    
+        }
+
+        [SerializeField]
+        [Tooltip("Max distance (in meters) to place an object if there is a raycast hit on a surface.")]
+        private float maxRaycastDistance = 20.0f;
+
+        /// <summary>
+        /// The max distance (in meters) to place an object if there is a raycast hit on a surface
+        /// </summary>
+        public float MaxRaycastDistance
+        {
+            get => maxRaycastDistance;
+            set => maxRaycastDistance = value;
+        }
+
+        [SerializeField]
+        [Tooltip("Array of LayerMask to execute from highest to lowest priority. First layermask to provide a raycast hit will be used by component.")]
+        private LayerMask[] magneticSurfaces = { UnityEngine.Physics.DefaultRaycastLayers };
+
+        /// <summary>
+        /// Array of LayerMask to execute from highest to lowest priority. First layermask to provide a raycast hit will be used by component.
+        /// </summary>
+        public LayerMask[] MagneticSurfaces
+        {
+            get => magneticSurfaces;
+            set => magneticSurfaces = value;
+        }
+
+        /// <summary>
+        /// Is the gameObjectToPlace currently in the placing state? This state is activated when the GameObjectToPlace
```

Comment:
Nit: "IsBeingPlaced is true when GameObjectToPlace is selected"

(i.e. just to be aware that some terms can be overloaded i.e. activated, also what does "state" refer to? - if the intent is to say that this property will have a true value when  GameObjectToPlace is "selected", it's simplest just to say that)

Reply:
Also, maybe "is selected" -> "is in the placing state"

I think that down below we use the term "placing state" - it's best to avoid introducing a bunch of terms which refer to the same concept and to be consistent on the usage of terms. Otherwise, does activated == selected == placing state?

Reply:
Ok, I agree that it is best to go with is selected and be consistent throughout the doc.

Reply:
Updated comments to "is selected" instead of "in the placing state"